server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/ScanDataSource.java

```diff
@@ -74,32 +74,26 @@
 
   ScanDataSource(Tablet tablet, Authorizations authorizations, byte[] defaultLabels, HashSet<Column> columnSet, List<IterInfo> ssiList,
       Map<String,Map<String,String>> ssio, AtomicBoolean interruptFlag, SamplerConfiguration samplerConfig, long batchTimeOut, String context) {
-    this.tablet = tablet;
-    expectedDeletionCount = tablet.getDataSourceDeletions();
-    this.options = new ScanOptions(-1, authorizations, defaultLabels, columnSet, ssiList, ssio, interruptFlag, false, samplerConfig, batchTimeOut, context);
-    this.interruptFlag = interruptFlag;
-    this.loadIters = true;
-    log.debug("new scan data source, tablet: {}, options: {}, interruptFlag: {}, loadIterators: {}", this.tablet, this.options, this.interruptFlag,
-        this.loadIters);
+    this(tablet, tablet.getDataSourceDeletions(), new ScanOptions(-1, authorizations, defaultLabels, columnSet, ssiList, ssio, interruptFlag, false,
+        samplerConfig, batchTimeOut, context), interruptFlag, true);
   }
 
   ScanDataSource(Tablet tablet, ScanOptions options) {
-    this.tablet = tablet;
-    expectedDeletionCount = tablet.getDataSourceDeletions();
-    this.options = options;
-    this.interruptFlag = options.getInterruptFlag();
-    this.loadIters = true;
-    log.debug("new scan data source, tablet: {}, options: {}, interruptFlag: {}, loadIterators: {}", this.tablet, this.options, this.interruptFlag,
-        this.loadIters);
+    this(tablet, tablet.getDataSourceDeletions(), options, options.getInterruptFlag(), true);
   }
 
   ScanDataSource(Tablet tablet, Authorizations authorizations, byte[] defaultLabels, AtomicBoolean iFlag) {
+    this(tablet, tablet.getDataSourceDeletions(), new ScanOptions(-1, authorizations, defaultLabels, EMPTY_COLS, null, null, iFlag, false, null, -1, null),
```

Comment:
it's unrelated but would it make sense to have a constructor for scan options that accepts only the configurable arguments in this ScanDataSource constructor? Other than this line it seems fine.